test/authrestapi_test.go

```diff
@@ -0,0 +1,223 @@
+		return true
+	} else {
+		log.Default().Println("Expected status code:", data.expected_status_code, "but got:", resp)
+		log.Default().Println("Data : ", data)
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Default().Println("Error reading response body:", err)
+			return false
+		}
+		log.Default().Println("Response body:", string(bodyBytes))
+		return false
+	}
+}
+
+func handlerForLogin(datai interface{}) bool {
+	data := datai.(Data)
+	url := data.url
+	jsonData := data.json_request
+
+	// Create a new HTTP request
+	req, err := http.NewRequest("POST", url, strings.NewReader(jsonData))
+	if err != nil {
+		log.Default().Println("Error creating request:", err)
+		return false
+	}
+
+	// Set the request content type to JSON
+	req.Header.Set("Content-Type", "application/json")
+
+	// Send the request
+	client := http.DefaultClient
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Default().Println("Error sending request:", err)
+		return false
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == int(data.expected_status_code) {
+		var loginResp model.LoginResponse
+		err = json.NewDecoder(resp.Body).Decode(&loginResp)
+		if err != nil {
+			log.Default().Println("Error decoding response:", err)
+			return false
+		}
+		if data.expected_status_code == 200 {
+			voucher = loginResp.Token
+			return true
+		} else {
+			log.Default().Println("Error in reponse :", loginResp.Error)
+			return true
+		}
+	} else {
+		log.Default().Println("Expected status code:", data.expected_status_code, "but got:", resp)
+		log.Default().Println("Data : ", data)
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Default().Println("Error reading response body:", err)
+			return false
+		}
+		log.Default().Println("Response body:", string(bodyBytes))
+		return false
+	}
+}
+
+func handlerForVerify(datai interface{}) bool {
+	data := datai.(Data)
+	url := data.url
+	jsonData := data.json_request
+
+	var verifyRequest model.VerifyVoucherRequest
+
+	err := json.Unmarshal([]byte(jsonData), &verifyRequest)
+
+	if err != nil {
+		log.Default().Println("Error decoding request for test:", err)
+		return false
+	}
+
+	if verifyRequest.Token == "proper" {
+		verifyRequest.Token = voucher
+	}
+	verifyRequestBytes, err := json.Marshal(verifyRequest)
+	if err != nil {
+		log.Default().Println("Error marshaling verifyRequest:", err)
+		return false
+	}
+	// Create a new HTTP request
+	req, err := http.NewRequest("POST", url, strings.NewReader(string(verifyRequestBytes)))
+	if err != nil {
+		log.Default().Println("Error creating request:", err)
+		return false
+	}
+
+	// Set the request content type to JSON
+	req.Header.Set("Content-Type", "application/json")
+
+	// Send the request
+	client := http.DefaultClient
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Default().Println("Error sending request:", err)
+		return false
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == int(data.expected_status_code) {
```

Comment:
For handling errors within unit test it would be better to use 'testify' library: https://tutorialedge.net/golang/improving-your-tests-with-testify-go/